components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3164,11 +3164,12 @@ where
             return;
         }
 
-        if self.fsm.peer.peer != *msg.get_to_peer() {
+        if self.fsm.peer.peer.get_id() != msg.get_to_peer().get_id() {
```

Comment:
Need to cover the case in tests.

Reply:
+1

Reply:
Sorry, I tried to write a test, but I don't know how to construct the scenario in issue #16504.

Reply:
@crazycs520 
I could add one when I have time.

Reply:
@Connor1996 
An integration test case is added, PTAL